test/framework/clusterctl/management_cluster.go

```diff
@@ -0,0 +1,118 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterctl
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterctlconfig "sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/discovery"
+)
+
+// Provides utilities for setting up a management cluster using clusterctl.
+
+// InitManagementClusterInput is the information required to initialize a new
+// management cluster for e2e testing.
+type InitManagementClusterInput struct {
+	// E2EConfig defining the configuration for the E2E test.
+	E2EConfig *E2EConfig
+
+	// ClusterctlConfigPath is the path to a clusterctl config file pointing to the repository to be used for running clusterctl init.
+	ClusterctlConfigPath string
+
+	// LogsFolder defines a folder where to store clusterctl logs.
+	LogsFolder string
+
+	// Scheme is used to initialize the scheme for the management cluster client.
+	Scheme *runtime.Scheme
+
+	// NewManagementClusterFn may be used to provide a custom function for
+	// returning a new management cluster. Otherwise kind.NewCluster is used.
+	NewManagementClusterFn func(name string, scheme *runtime.Scheme) (cluster framework.ManagementCluster, kubeConfigPath string, err error)
+}
+
+// InitManagementCluster returns a new cluster initialized and the path to the kubeConfig file to be used to access it.
+func InitManagementCluster(ctx context.Context, input *InitManagementClusterInput) (framework.ManagementCluster, string) {
+	// validate parameters and apply defaults
+
+	Expect(input.E2EConfig).ToNot(BeNil(), "Invalid argument. input.E2EConfig can't be nil when calling InitManagementCluster")
+	Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Failed to create a clusterctl config file at %s", input.ClusterctlConfigPath)
+
+	By(fmt.Sprintf("Creating the management cluster with name %s", input.E2EConfig.ManagementClusterName))
+
+	managementCluster, managementClusterKubeConfigPath, err := input.NewManagementClusterFn(input.E2EConfig.ManagementClusterName, input.Scheme)
```

Comment:
Do we need to add a check here in case NewManagementClusterFn is nil?

Are we using kind.NewCluster as in the NewManagementClusterFn comment:

	// NewManagementClusterFn may be used to provide a custom function for
	// returning a new management cluster. Otherwise kind.NewCluster is used.

Reply:
fixed validation